interop/xds/server/server.go

```diff
@@ -78,8 +93,76 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}
+
+	for _, headerVal := range getRPCBehaviorMetadata(ctx) {
+		if strings.HasPrefix(headerVal, sleepPfx) {
+			sleep, err := strconv.Atoi(headerVal[len(sleepPfx):])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header: %v", headerVal)
```

Comment:
Here and everywhere else, where you return an error, please consider returning `nil` for the response.

Reply:
In the cpp implementation they are setting the server ID in the response for even error returns, so I had followed the same approach
will correct it now